src/Component/Homepage.js

```diff
@@ -0,0 +1,218 @@
+        localStorage.clear();
+        navigate('/');
+      });
+  }
+
+  // Generate RefreshToken Using AuthCode
+  async function generateRefreshToken(AuthCode) {
+    let data = new FormData();
+    data.append('code', AuthCode);
+    data.append('client_id', process.env.REACT_APP_CLIENT_KEY);
+    data.append('client_secret', process.env.REACT_APP_CLIENT_SECRET);
+    data.append('grant_type', 'authorization_code');
+    data.append('redirect_uri', 'http://localhost:3000');
+
+    let config = {
+      method: 'post',
+      url: 'https://oauth2.googleapis.com/token',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      data,
+    };
+
+    await axios(config)
+      .then(async (response) => {
+        localStorage.setItem('Access_Token', response.data.access_token);
+        localStorage.setItem('Refresh_Token', response.data.refresh_token);
+        localStorage.setItem('Gen_Date', new Date().getTime());
+      })
+      .catch(() => {
+        localStorage.clear();
+        navigate('/');
+      });
+
+    const Access_Token = localStorage.getItem('Access_Token');
+
+    data = new FormData();
+    config = {
+      method: 'get',
+      url: `https://www.googleapis.com/oauth2/v1/userinfo?alt=json&access_token=${Access_Token}`,
+    };
+
+    await axios(config).then(async (response) => {
+      localStorage.setItem('first_name', response.data.given_name);
+      localStorage.setItem('last_name', response.data.family_name);
+      localStorage.setItem('email', response.data.email);
+
+      data = new FormData();
+      data.append('first_name', localStorage.getItem('first_name'));
+      data.append('last_name', localStorage.getItem('last_name'));
+      data.append('username', localStorage.getItem('first_name'));
+      data.append('email', localStorage.getItem('email'));
+
+      config = {
+        method: 'post',
+        url: '/employees/employees/',
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+        data,
+      };
+
+      await axios(config)
+        .then(async (res) => {
+          localStorage.setItem('user_id', res.data.detail.id);
+          localStorage.setItem('django_Token', res.data.Token);
+        })
+        .catch(() => {
+          navigate('/');
+        });
+    });
+  }
+
+  // After Successfully Login
+  const onSuccess = async (res) => {
+    const Auth = res.code;
+    await generateRefreshToken(Auth);
+
+    navigate('/dashboard');
+  };
+
+  const onFailure = () => {};
+
+  useEffect(() => {
+    const Refresh_Token = localStorage.getItem('Refresh_Token');
+    if (Refresh_Token == null) {
+      return;
+    }
+
+    let chk = true;
+    const ExcessTokenExpiry =
+      new Date().getTime() - localStorage.getItem('Gen_Date');
+
+    if (ExcessTokenExpiry >= 3500000) {
+      chk = false;
+    }
+
+    if (!chk) {
+      generateAccessToken(Refresh_Token);
+    }
+
+    navigate('/Dashboard');
+  });
```

Comment:
How will this method behave?

Reply:
When user land on this page first run this method so if user session is stored then it redirect to dashboard.